gateway-shell/src/test/java/org/apache/knox/gateway/shell/knox/token/TokenTest.java

```diff
@@ -76,4 +127,100 @@ private void testToken(boolean setDoAsUser, String doAsUser) {
 
     verify(knoxSession);
   }
+
+  private void testRenewToken(boolean setDoAsUser, String doAsUser) throws Exception {
+    final String testToken = "ABCDEFG123456";
+
+    KnoxSession knoxSession = createMock(KnoxSession.class);
+    expect(knoxSession.base()).andReturn("http://localhost/base").atLeastOnce();
+    expect(knoxSession.getHeaders()).andReturn(Collections.emptyMap()).atLeastOnce();
+    expect(knoxSession.executeNow(isA(HttpRequest.class))).andReturn(null).atLeastOnce();
+    replay(knoxSession);
+
+    Renew.Request request = (setDoAsUser)
+        ? Token.renew(knoxSession, testToken, doAsUser)
+        : Token.renew(knoxSession, testToken);
+
+    boolean shouldHaveDoAs = false;
+
+    if (setDoAsUser) {
+      assertEquals(doAsUser, request.getDoAsUser());
+      shouldHaveDoAs = StringUtils.isNotEmpty(doAsUser);
+    } else {
+      assertNull(request.getDoAsUser());
+    }
+
+    assertEquals("http://localhost/base/knoxtoken/api/v1/token/renew" + (shouldHaveDoAs ? ("?doAs=" + doAsUser) : ""),
+                 request.getRequestURI().toString());
+
+    assertEquals(testToken, request.getToken());
+
+    Callable<TokenLifecycleResponse> callable = request.callable();
+    try {
+      callable.call();
+    } catch (Exception e) {
+      e.printStackTrace();
```

Comment:
Should this fail the test? (same with other catch locations)

Reply:
Actually, I should remove the printing of the stack trace. The call should fail, but is required to setup the actual underlying request (so the setting of the entity can be verified). Short answer: no, it should not fail the test in either case.